src/ATen/native/xpu/BatchLinearAlgebra.cpp

```diff
@@ -0,0 +1,63 @@
+#include <ATen/core/Tensor.h>
+#include <ATen/native/BatchLinearAlgebra.h>
+#include <ATen/native/DispatchStub.h>
+#include <ATen/ops/empty_like.h>
+#if defined(USE_ONEMKL)
+#include <ATen/native/xpu/mkl/BatchLinearAlgebra.h>
+#endif // USE_ONEMKL
+
+namespace at::native {
+
+void svd_kernel_xpu(
+    const Tensor& A,
+    const bool full_matrices,
+    const bool compute_uv,
+    const c10::optional<c10::string_view>& driver,
+    const Tensor& U,
+    const Tensor& S,
+    const Tensor& Vh,
+    const Tensor& info) {
+#if defined(USE_ONEMKL)
+  native::xpu::svd_mkl(A, full_matrices, compute_uv, driver, U, S, Vh, info);
+#else
+  const auto A_cpu = A.to(A.options()
+                              .device(kCPU)
+                              .memory_format(at::MemoryFormat::Contiguous)
+                              .pinned_memory(true));
+  // U, S, Vh, info are the right size and strides, but are on GPU
+  // We copy them into CPU in pinned_memory
+  const auto empty_like_cpu = [](const Tensor& t) {
+    return at::empty_like(t, t.options().device(kCPU).pinned_memory(true));
```

Comment:
We don't use pin memory for CPU fallback? Allocating too much pin memory will lag the performance.

Reply:
Pinned memory has been removed.